lib/src/base_logger.dart

```diff
@@ -0,0 +1,7 @@
+
+abstract class BaseLogger {
```

Comment:
I believe that 'EventLogger' would be a more appropriate name for this interface than 'BaseLogger', as it is an abstract class that other classes will conform to rather than inherit from, much like a protocol.

below is what gpt said to me.
---
Yes, you are correct. Using the name "EventLogger" for your interface is a better choice than "BaseLogger", especially if your interface is an abstract class that is intended to be implemented by other classes.

The name "BaseLogger" implies that it is a base class that provides common functionality to other classes, but in this case, your interface is not intended to provide any implementation details. Instead, it defines a set of methods that other classes must implement to conform to the interface.

On the other hand, the name "EventLogger" accurately describes the purpose of the interface, which is to define a set of methods for logging events. This name is more descriptive and intuitive, and makes it clear to other developers what the interface is intended to be used for.

So, I would recommend renaming your interface from "BaseLogger" to "EventLogger" to make it more clear and intuitive to other developers.